test/e2e/os/amazonlinux.go

```diff
@@ -19,34 +19,34 @@ type amazonLinuxCloudInitData struct {
 }
 
 type AmazonLinux2023 struct {
-	Architecture string
+	amiArchitecture string
+	architecture    architecture
 }
 
 func NewAmazonLinux2023AMD() *AmazonLinux2023 {
 	al := new(AmazonLinux2023)
-	al.Architecture = x8664Arch
+	al.amiArchitecture = x8664Arch
+	al.architecture = amd64
 	return al
 }
 
 func NewAmazonLinux2023ARM() *AmazonLinux2023 {
 	al := new(AmazonLinux2023)
-	al.Architecture = arm64Arch
+	al.amiArchitecture = arm64Arch
+	al.architecture = arm64
 	return al
 }
 
 func (a AmazonLinux2023) Name() string {
-	if a.Architecture == x8664Arch {
-		return "al23-amd64"
-	}
-	return "al23-arm64"
+	return "al23-" + a.architecture.String()
```

Comment:
Just a nit that I missed during previous reviews, but any reason why we shouldn't change all of these to `Sprintf`s? Or maybe even a `generateName` variadic function whose arguments you can append to the OS name.

Reply:
For simple string concatenation, `+` is usually better (IMO): more expressive, simpler and easier to read. As a bonus, it's also *faster*, but tbh that's not a concern here.

About having a function for name generation, I can see that if we want to standarize the OS name format. However, I'm not sure that's what we nee here. In fact, they are already not standard since some of them have the `-docker` suffix. I'm more inclined to leave it as it is.